src/AutoRest.CSharp/Common/Input/InputTypes/InputModelType.cs

```diff
@@ -4,17 +4,19 @@
 using System;
 using System.Collections.Generic;
 using System.Diagnostics;
+using System.Linq;
 using System.Runtime.CompilerServices;
 
 namespace AutoRest.CSharp.Common.Input
 {
-    internal record InputModelType(string Name, string? Namespace, string? Accessibility, string? Deprecated, string? Description, InputModelTypeUsage Usage, IReadOnlyList<InputModelProperty> Properties, InputModelType? BaseModel, IReadOnlyList<InputModelType> DerivedModels, string? DiscriminatorValue, string? DiscriminatorPropertyName, InputDictionaryType? InheritedDictionaryType, bool IsNullable)
-        : InputType(Name, IsNullable)
+    internal record InputModelType(string Name, string? Namespace, string? Accessibility, string? Deprecated, string? Description, InputModelTypeUsage Usage, IReadOnlyList<InputModelProperty> Properties, InputModelType? BaseModel, IReadOnlyList<InputModelType> DerivedModels, string? DiscriminatorValue, string? DiscriminatorPropertyName, InputDictionaryType? InheritedDictionaryType, bool IsNullable, string[] SerializationFormats, bool IsEmpty = false, string? OriginalName = null)
```

Comment:
SerializationFormats needs discussion with typespec team
IsEmpty can be replaced
`Original` should be set to `Name` in the constructor and have a getter for it, remove this from constructor, rename this to `SpecName`

Reply:
IReadonlyList<string>